parsec/api/protocol/base.py

```diff
@@ -80,13 +80,23 @@ class RequireGreaterTimestampRepSchema(BaseRepSchema):
 
 
 class TimestampOutOfBallparkRepSchema(BaseRepSchema):
+    """This schema has been added to API version 2.4.
+
+    However, it re-uses the `bad_timestamp` status that was used for similar errors in previous backend versions.
+    For compatibility purposes, this schema should be compatible with `ErrorRepSchema` in the sense that:
+    - an `ErrorRepSchema` with status `bad_timestamp` should be able to deserialize into a `TimestampOutOfBallparkRepSchema`
+    - a `TimestampOutOfBallparkRepSchema` should be able to deserialize into an `ErrorRepSchema` with status `bad_timestamp
+
+    New clients who wishes to use those fields should check for their existence first.
+    """
+
     # `bad_timestamp` is kept for backward compatibility,
     # even though `timestamp_out_of_ballpark` would be more explicit
     status: fields.CheckedConstant = fields.CheckedConstant("bad_timestamp", required=True)
-    ballpark_client_early_offset = fields.Float(required=True, allow_none=False)
-    ballpark_client_late_offset = fields.Float(required=True, allow_none=False)
-    client_timestamp = fields.DateTime(required=True, allow_none=False)
-    backend_timestamp = fields.DateTime(required=True, allow_none=False)
+    ballpark_client_early_offset = fields.Float(required=False, missing=None)
+    ballpark_client_late_offset = fields.Float(required=False, missing=None)
+    client_timestamp = fields.DateTime(required=False, missing=None)
+    backend_timestamp = fields.DateTime(required=False, missing=None)
```

Comment:
Can't we use a `@pre_load` schema decorator instead ? This way all the legacy handling would be contained there instead of being scattered across the codebase.

On top of that it will be much easier to remove this corner case in the future once the versions<2.4 become old enoughh

Reply:
After more discussion with @vxgmichel, it appear those values cannot be easily generated if they are missing: this would imply choosing an arbitrary ballpark, then a `client_timestamp` (while the client may have kept the original `client_timestamp`...) and do a cheating aritmetic so that the `backend_timestamp` makes things out of ballpark :/